app/src/features/apiClient/helpers/auth.ts

```diff
@@ -0,0 +1,68 @@
+import { isEmpty, unionBy } from "lodash";
+import { AUTHORIZATION_TYPES } from "../screens/apiClient/components/clientView/components/request/components/AuthorizationView/authStaticData";
+import { AUTH_ENTRY_IDENTIFIER } from "../screens/apiClient/components/clientView/components/request/components/AuthorizationView/types";
+import { KeyValuePair, RQAPI } from "../types";
+
+export const processAuthOptions = (authOptions: RQAPI.AuthOptions) => {
+  const headers: KeyValuePair[] = [];
+  const queryParams: KeyValuePair[] = [];
+
+  const { currentAuthType = "" } = authOptions;
+
+  let newKeyValuePair: KeyValuePair;
+
+  const createAuthorizationKeyValuePair = (type: string, key: string, value: string): typeof newKeyValuePair => ({
+    id: Date.now(),
+    key,
+    value,
+    type,
+    isEnabled: true,
+  });
+
+  const updateDataInState = (data: KeyValuePair[], key: string, value: string) => {
+    const existingIndex = data.findIndex((option) => option.type === AUTH_ENTRY_IDENTIFIER);
```

Comment:
This looks redundant as the entries in the data corresponds to AuthEntries only. Instead we should check if the data contains an entry with same key

Reply:
this was done to replicate the postman behaviour postman allows user to add the authorization header manually so if we go by the key then it will change the header added by user not by us using auhtorization tab.
[
![Screenshot 2024-12-19 at 12 33 38 PM](https://github.com/user-attachments/assets/5070afd1-9c2e-4a48-8341-31de21924fc2)
](url)

Reply:
there can only be one auth header so that's why we can reply on key